### PR TITLE
Fix compilation when HAL_WITH_ESC_TELEM == 0

### PR DESCRIPTION
**Pull request by amilcarlucas**

### CODE REVIEW

**Review by andyp1per — CHANGES_REQUESTED** [review]
I don't think this is right. The various ESC drivers cannot exist without ESC telemetry - they did not do so before and should not do so now. How are you provoking this?

File: libraries/AP_ESC_Telem/AP_ESC_Telem_Backend.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+
+#ifndef HAL_WITH_ESC_TELEM
+#define HAL_WITH_ESC_TELEM 1
```
Comment (amilcarlucas) [inline libraries/AP_ESC_Telem/AP_ESC_Telem_Backend.h]: This PR fixes compilation when `#define HAL_WITH_ESC_TELEM 0` is used here.

**Comment by amilcarlucas** [comment]
Yes they can, without telemetry they just control the motors, and are still useful.
For some models like the skyviper this will save some memory, besides that it fixes the fact that you currently can not set HAL_WITH_ESC_TELEM to zero in the libraries/AP_ESC_Telem/AP_ESC_Telem_Backend.h file

Why is the HAL_WITH_ESC_TELEM  macro there if it can not be changed?

**Comment by andyp1per** [comment]
@amilcarlucas I think the flag should be forced to 1 if any of the ESC drivers are present

**Comment by amilcarlucas** [comment]
it reduces a CubeOrange build size by 7524 bytes. If the SDcard logging functions get moved into the AP_ESC_Telem directory it will save even more.

No I do not think the ESC drivers should automatically enable it. If the user wants to disable it, he/she should be able to do it!
And this PR allows just that.

**Comment by amilcarlucas** [comment]
All CI tests pass with HAL_WITH_ESC_TELEM == 0

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -6939,11 +6939,6 @@ def tests2b(self):  # this block currently around 9.5mins here
                  "Ensure position doesn't zero when GPS lost",
                  self.test_copter_gps_zero),
 
-            Test("DynamicRpmNotches",
```
Comment (andyp1per) [inline Tools/autotest/arducopter.py]: Why have you removed this?

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 3]
```diff
@@ -244,10 +244,6 @@ singleton AP_Baro method get_pressure float
 singleton AP_Baro method get_temperature float
 singleton AP_Baro method get_external_temperature float
 
-include AP_ESC_Telem/AP_ESC_Telem.h
```
Comment (andyp1per) [inline libraries/AP_Scripting/generator/description/bindings.desc:249]: Why have you removed this?

**Comment by andyp1per** [comment]
@amilcarlucas if we do want to make this optional - and I am still unconvinced as it adds complexity and I really can't see why you would want to compile it out, then I probably won't use inheritance for the backed but make it a member variable

File: Tools/autotest/arducopter.py (hunk 2 above)
Comment (amilcarlucas) [reply Tools/autotest/arducopter.py]: This test obviously will not pass when HAL_WITH_ESC_TELEM == 0. this last commit should not get merged, it is just there to get CI to run.

File: libraries/AP_Scripting/generator/description/bindings.desc (hunk 3 above)
Comment (amilcarlucas) [reply libraries/AP_Scripting/generator/description/bindings.desc:249]: Again, this is just to make CI pass, this is part of the last commit, that should not get merged.

**Comment by amilcarlucas** [comment]
I do not feel it adds that much complexity. And boards like SkyViper will surely enjoy not having the extra 7524 bytes for a feature they they can not use.

**Comment by andyp1per** [comment]
Why can't the SkyViper use BLHeli32?

**Comment by amilcarlucas** [comment]
Because it uses [brushed motors and the ESCs are integrated in the flight controller](https://ardupilot.org/copter/docs/skyrocket.html).

**Comment by amilcarlucas** [comment]
Compiled it for a CubeOrange-bdshot board, and it compiles fine. So it is possible to set HAL_WITH_ESC_TELEM == 0 in every possible scenario!

This PR does not open the padora box, it closes it :)

**Comment by andyp1per** [comment]
So is it true to say therefore that the skyviper does not support dshot either?

**Comment by amilcarlucas** [comment]
yes

**Comment by andyp1per** [comment]
Ok I have fixed this a different way. On the skyviper you should set HAL_SUPPORT_RCOUT_SERIAL to 0 and it will give you what you want.

**Comment by amilcarlucas** [comment]
I rebased this one, just in case.

**Pull request closed without merging**